frontend/src/components/unsubscribe/Unsubscribe.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useState } from 'react'
+import { useParams } from 'react-router-dom'
+import querystring from 'querystring'
+
+import { ErrorBlock, PrimaryButton, TextButton } from 'components/common'
+
+import styles from './Unsubscribe.module.scss'
+import appLogo from 'assets/img/brand/app-logo.svg'
+import landingHero from 'assets/img/unsubscribe/request-unsubscribe.png'
+import cancelRequestHero from 'assets/img/unsubscribe/cancel-request.png'
+
+import { unsubscribeRequest } from 'services/unsubscribe.service'
+
+const Unsubscribe = () => {
+  const { version } = useParams()
+  const [errorMsg, setErrorMsg] = useState('')
+  const [isUnsubscribed, setUnsubscribed] = useState(false)
+  const [isStaying, setStaying] = useState(false)
+
+  async function onConfirmation() {
+    try {
+      const query = new URL(window.location.href).searchParams.toString()
+      const params = querystring.parse(query)
+
+      // Check to make sure that the search params are all strings and not []strings
+      const isValid = Object.values(params).every((v) => typeof v === 'string')
+      if (!isValid) {
+        throw new Error(
+          'Search params should all be strings and not array of strings.'
+        )
```

Comment:
This error message would not be understandable by users, I think it would be sufficient to throw a general error something like `Invalid subscribe request` or `Error occured during unsubscription`